### PR TITLE
TSK-1469: Update to Angular 11

### PR DESCRIPTION
**Pull request by Tristan2357**
---
 
### For the submitter:
- [x] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [x] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [x] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Comment by Tristan2357** [comment]
Documentation and release notes will be added once this pr is approved to avoid confusion

File: web/src/app/app.component.html
Comment (sofie29) [inline web/src/app/app.component.html:27]: With this change, the space for the progress bar is deleted as soon as the progress bar disappears. Consequently, the whole view shifts a little bit down and up. Please find another solution.

File: web/src/app/monitor/components/task-report/task-report.component.html
Comment (sofie29) [inline web/src/app/monitor/components/task-report/task-report.component.html:6]: Is this class needed? I can't see a usage.
In general, we are using BEM class convention http://getbem.com/naming/ and the class name should start with the name of the component. So `monitor-chart` should be `task-report__monitor-chart` in case you need the class.

File: web/src/app/app.component.html
Comment (sofie29) [reply web/src/app/app.component.html:27]: Additionally, the progress bar doesn't disappear after starting the backend and opening the app as soon as the backend is ready.

File: web/src/app/shared/components/user-information/user-information.component.spec.ts
Comment (sofie29) [inline web/src/app/shared/components/user-information/user-information.component.spec.ts:8]: this import isn't needed anymore :)

File: web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss
Comment (sofie29) [inline web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss:47]: There already is a css class named `additional-toolbar` see line 17.

File: web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss
Comment (sofie29) [inline web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss:41]: Is there a usage? I can't see it.

File: web/src/theme/_main.scss
Comment (sofie29) [inline web/src/theme/_main.scss:1]: We replaced the bootstrap-datepicker with the datepicker from angular material, so I'm not sure whether we need this import. However, removing all files belonging to the unused bootstrap-datepicker can be done in separate ticket as well.

File: web/src/theme/bootstrap-replacement.scss
Comment (sofie29) [inline web/src/theme/bootstrap-replacement.scss:1]: Can you explain why we need this class? I don't understand :)

**Review by sofie29 — CHANGES_REQUESTED** [review]

**Review by sofie29 — CHANGES_REQUESTED** [review]
When you open a task in the workplace, select `custom attributes` or `callback information`, you can see that button and input field are not at the same position in vertical direction anymore. Please fix. It is the `task-attribute-value` component.

File: web/src/theme/bootstrap-replacement.scss
Comment (Tristan2357) [reply web/src/theme/bootstrap-replacement.scss:1]: I created this file to replace the defaults, which bootstrap had and our frontend relied on.
These include basic things, like the size and spacing of some headings, and some other low level styling.
I thought about styling elements individually or putting this css in one of the other global styling files, but to be honest, i do not quite understand our global-css file structure.

File: web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss
Comment (Tristan2357) [reply web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss]: Removed

File: web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss
Comment (Tristan2357) [reply web/src/app/workplace/components/task-list-toolbar/task-list-toolbar.component.scss]: Merged

File: web/src/app/monitor/components/task-report/task-report.component.html
Comment (Tristan2357) [reply web/src/app/monitor/components/task-report/task-report.component.html]: Removed

File: web/src/app/app.component.html
Comment (Tristan2357) [reply web/src/app/app.component.html]: Replaced with an *ngIf, as [hidden] did not appear to work

**Comment by Tristan2357** [comment]
Some css change affected the layout of the custom attributes, this should be fixed in the newest commit

File: web/src/app/app.component.html
Comment (sofie29) [reply web/src/app/app.component.html]: The new solution causes `ExpressionChangedAfterItHasBeenCheckedError ` every time the progress bar is shown.

**Comment by sofie29** [comment]
> When you open a task in the workplace, select `custom attributes` or `callback information`, you can see that button and input field are not at the same position in vertical direction anymore. Please fix. It is the `task-attribute-value` component.

This doesn't seem to be fixed.

**Review by cnguyen-de — APPROVED** [review]
looks good, wait for new release to merge into master

**Review by sofie29 — DISMISSED** [review]

**Review by sofie29 — APPROVED** [review]

**Pull request merged**
